libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,22 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async finishBook(
+    id: string,
+    finishedDate: string
+  ): Promise<ReadingListItem[]> {
+    this.storage.update(list =>
```

Comment:
This can be rewritten as below:

```
this.storage.update(list => {
      return list.map(book => {
           return book.bookId !== id ? book : { ...book, finishedDate, finished: true };
     });
});
```


if you use above code then you don't need this line and line no 47. You can remove them and just keep above code snippet.